### PR TITLE
Change Slot type to be non-nullable by default

### PR DESCRIPTION
**Pull request by behowell**
## Current Behavior

The `Slot` type has a built-in `| null`, which makes it less obvious that most slots allow being set to `null`, which would cause them not to be rendered. There are a few cases in our library where this appears to be a mistake (the slot was supposed to be NonNullable, but that was omitted).

The `ComponentState` type also had a special-case for the `root` slot, never allowing it to be null. Although the root slot never should be null, that should be explicit in the definition of the root slot, not implicit through a special case in `ComponentState`.

### Update slot utility types in react-utilities/src/compose:
* types.ts:
  * Remove `| null` from the `Slot` type
  * Remove the special case for `root` in the `ComponentState` type
* getSlots.ts:
  * Fix return types for `getSlots`: the slot should be undefined if its type includes `null` or the type is optional (`?`)
  * Have it return `undefined` for missing slots instead of `null`
* resolveShorthand.ts:
  * Clean up the return type so it includes `undefined` only if the slot type includes `undefined` (aka is optional `?`)

### Update slot types in all other packages:
* Replace `NonNullable<Slot<...>>` with `Slot<...>` because NonNullable is now the default.
* Replace  `Slot<...>` with `Slot<...> | null` since they were previously nullable by default.
  * EXCEPT: if a slot was previously not optional (`?`), I did not add `| null` because it appears that it was a mistake and should have been NonNullable in the first place.

### CODE REVIEW

File: packages/react-accordion/src/components/AccordionHeader/AccordionHeader.types.ts [hunk 1]
```diff
@@ -23,15 +23,15 @@ export type AccordionHeaderSlots = {
   /**
    * The component to be used as button in heading
    */
-  button: NonNullable<Slot<ARIAButtonSlotProps>>;
+  button: Slot<ARIAButtonSlotProps>;
   /**
    * Expand icon slot rendered before (or after) children content in heading
    */
   expandIcon: Slot<'span'>;
```
Comment (behowell) [inline packages/react-accordion/src/components/AccordionHeader/AccordionHeader.types.ts:30]: Note: this slot was previously nullable, but with this change it is non-nullable (as a result of `Slot` no longer including `| null`). It appears from the code that it was never intended to be nullable.

File: packages/react-card/src/components/CardHeader/CardHeader.types.ts [hunk 2]
```diff
@@ -3,10 +3,10 @@ import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utili
 export type CardHeaderSlots = {
   root: Slot<'div'>;
   image: Slot<'div'>;
-  content?: Slot<'div'>;
+  content?: Slot<'div'> | null;
   header: Slot<'span'>;
   description: Slot<'span'>;
```
Comment (behowell) [inline packages/react-card/src/components/CardHeader/CardHeader.types.ts:8]: Note: the `image`, `header`, and `description` slots were previously nullable, but with this change they are non-nullable (as a result of `Slot` no longer including `| null`). It appears from the code that they were never intended to be nullable.

File: packages/react-checkbox/src/components/Checkbox/Checkbox.types.ts [hunk 3]
```diff
@@ -49,20 +49,20 @@ export type CheckboxSlots = {
...
    * Hidden input that handles the checkbox's functionality.
    *
    * This is the PRIMARY slot: all native properties specified directly on `<Checkbox>` will be applied to this slot,
    * except `className` and `style`, which remain on the root slot.
    */
-  input: NonNullable<Slot<'input'>>;
+  input: Slot<'input'>;
 
   /**
    * The checkbox, with the checkmark icon as its child when checked.
```
Comment (behowell) [inline packages/react-checkbox/src/components/Checkbox/Checkbox.types.ts:68]: Note: the `indicator` slot was previously nullable, but with this change it is non-nullable (as a result of `Slot` no longer including `| null`). It appears from the code that it was never intended to be nullable.

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/d8ab052dcb2a452d879f9a01ae8abd30fd44f57b'>d8ab052dcb2a452d879f9a01ae8abd30fd44f57b</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Breadcrumb-mount_base.html">2717</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Breadcrumb-mount.html">2283</a></td>
            <td>1000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Breadcrumb-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/FluentProviderWithTheme-mount_base.html">156</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/FluentProviderWithTheme-mount.html">118</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/FluentProviderWithTheme-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Avatar-mount_base.html">766</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Avatar-mount.html">764</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/BaseButton-mount_base.html">813</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/BaseButton-mount.html">805</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Breadcrumb-mount_base.html">2717</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Breadcrumb-mount.html">2283</a></td>
            <td>1000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Breadcrumb-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>ButtonNext</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/ButtonNext-mount_base.html">425</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/ButtonNext-mount.html">425</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Checkbox-mount_base.html">1334</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Checkbox-mount.html">1918</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/CheckboxBase-mount_base.html">1116</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/CheckboxBase-mount.html">1112</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/ChoiceGroup-mount_base.html">4131</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/ChoiceGroup-mount.html">4144</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/ComboBox-mount_base.html">874</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/ComboBox-mount.html">863</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/CommandBar-mount_base.html">8858</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/CommandBar-mount.html">8844</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/ContextualMenu-mount_base.html">7350</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/ContextualMenu-mount.html">7415</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/DefaultButton-mount_base.html">985</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/DefaultButton-mount.html">995</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/DetailsRow-mount_base.html">3229</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/DetailsRow-mount.html">3245</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/DetailsRowFast-mount_base.html">3202</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/DetailsRowFast-mount.html">3248</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/DetailsRowNoStyles-mount_base.html">3047</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/DetailsRowNoStyles-mount.html">3074</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Dialog-mount_base.html">1940</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Dialog-mount.html">1944</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/DocumentCardTitle-mount_base.html">154</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/DocumentCardTitle-mount.html">165</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Dropdown-mount_base.html">2805</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Dropdown-mount.html">2763</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderNext</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/FluentProviderNext-mount_base.html">1760</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/FluentProviderNext-mount.html">1643</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/FluentProviderWithTheme-mount_base.html">156</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/FluentProviderWithTheme-mount.html">118</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/FluentProviderWithTheme-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/FluentProviderWithTheme-virtual-rerender_base.html">108</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/FluentProviderWithTheme-virtual-rerender.html">105</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">167</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/FluentProviderWithTheme-virtual-rerender-with-unmount.html">172</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/FocusTrapZone-mount_base.html">1605</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/FocusTrapZone-mount.html">1618</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/FocusZone-mount_base.html">1560</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/FocusZone-mount.html">1576</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/IconButton-mount_base.html">1533</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/IconButton-mount.html">1536</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Label-mount_base.html">322</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Label-mount.html">331</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Layer-mount_base.html">2559</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Layer-mount.html">2580</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Link-mount_base.html">429</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Link-mount.html">434</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/MakeStyles-mount_base.html">1486</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/MakeStyles-mount.html">1514</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/MenuButton-mount_base.html">1312</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/MenuButton-mount.html">1282</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/MessageBar-mount_base.html">1750</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/MessageBar-mount.html">1781</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Nav-mount_base.html">2838</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Nav-mount.html">2857</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/OverflowSet-mount_base.html">975</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/OverflowSet-mount.html">976</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Panel-mount_base.html">1857</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Panel-mount.html">1864</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Persona-mount_base.html">758</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Persona-mount.html">740</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Pivot-mount_base.html">1247</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Pivot-mount.html">1255</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/PrimaryButton-mount_base.html">1134</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/PrimaryButton-mount.html">1124</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Rating-mount_base.html">6669</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Rating-mount.html">6629</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/SearchBox-mount_base.html">1159</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/SearchBox-mount.html">1185</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Shimmer-mount_base.html">2247</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Shimmer-mount.html">2193</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Slider-mount_base.html">1730</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Slider-mount.html">1716</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/SpinButton-mount_base.html">4310</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/SpinButton-mount.html">4298</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Spinner-mount_base.html">392</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Spinner-mount.html">388</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/SplitButton-mount_base.html">2718</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/SplitButton-mount.html">2691</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Stack-mount_base.html">473</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Stack-mount.html">469</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/StackWithIntrinsicChildren-mount_base.html">2038</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/StackWithIntrinsicChildren-mount.html">2023</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/StackWithTextChildren-mount_base.html">4575</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/StackWithTextChildren-mount.html">4545</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/SwatchColorPicker-mount_base.html">9894</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/SwatchColorPicker-mount.html">9930</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/TagPicker-mount_base.html">2256</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/TagPicker-mount.html">2290</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/TeachingBubble-mount_base.html">11363</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/TeachingBubble-mount.html">11387</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Text-mount_base.html">381</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Text-mount.html">400</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/TextField-mount_base.html">1206</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/TextField-mount.html">1217</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/ThemeProvider-mount_base.html">1026</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/ThemeProvider-mount.html">1043</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/ThemeProvider-virtual-rerender_base.html">543</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/ThemeProvider-virtual-rerender.html">544</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">1627</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">1615</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Toggle-mount_base.html">727</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/Toggle-mount.html">737</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/buttonNative-mount_base.html">127</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21914/perf-test/buttonNative-mount.html">137</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

File: packages/react-select/src/components/Select/Select.types.ts [hunk 4]
```diff
@@ -6,13 +6,13 @@ export type SelectSlots = {
...
-  root: NonNullable<Slot<'span'>>;
+  root: Slot<'span'>;
 
   /** Primary slot: the actual `<select>` element */
-  select: NonNullable<Slot<'select'>>;
+  select: Slot<'select'>;
 
   /** the icon, typically a down arrow */
-  icon: Slot<'span'>;
+  icon?: Slot<'span'> | null;
```
Comment (behowell) [inline packages/react-select/src/components/Select/Select.types.ts:15]: Need to revert this: I don't think this slot was meant to be nullable even though it previously was
```suggestion
  icon: Slot<'span'>;
```

File: packages/react-select/src/components/Select/renderSelect.tsx [hunk 5]
```diff
@@ -10,7 +10,7 @@ export const renderSelect_unstable = (state: SelectState) => {
   return (
     <slots.root {...slotProps.root}>
       <slots.select {...slotProps.select}>{slotProps.select.children}</slots.select>
-      <slots.icon {...slotProps.icon} />
+      {slots.icon && <slots.icon {...slotProps.icon} />}
```
Comment (behowell) [inline packages/react-select/src/components/Select/renderSelect.tsx:13]: Need to revert this as well:
```suggestion
      <slots.icon {...slotProps.icon} />
```

**Pull request closed without merging**